pkg/client/api.go

```diff
@@ -227,7 +276,14 @@ func (h *queryClient) QueryRange(ctx context.Context, r Range, query Selector) (
 	}
 
 	var queryRes QueryResult
-	err = json.Unmarshal(res.Data, &queryRes)
+	if err := json.Unmarshal(res.Data, &queryRes); err != nil {
+		return queryRes, &Error{
+			Type:        ErrBadData,
+			ErrorMsg:    fmt.Sprintf("failed to unmarshal JSON response: %v", err),
+			StatusCode:  res.StatusCode, // Use the status code from the response
```

Comment:
ditto: keep the same err.StatusCode and let the increment in the failed query metrics counter show that this, even if a 200 OK, was an error